tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -140,22 +150,32 @@ class IncrementalAssignmentConsumerEventHandler(ConsumerEventHandler):
     def __init__(self, node, verify_offsets, idx):
         super().__init__(node, verify_offsets, idx)
         
-    def handle_partitions_revoked(self, event):
+    def handle_partitions_revoked(self, event, node, logger):
         self.revoked_count += 1
         self.state = ConsumerState.Rebalancing
         self.position = {}
+        revoked = []
+
         for topic_partition in event["partitions"]:
-            topic = topic_partition["topic"]
-            partition = topic_partition["partition"]
-            self.assignment.remove(TopicPartition(topic, partition))
+            tp = _create_partition_from_dict(topic_partition)
 
-    def handle_partitions_assigned(self, event):
+            if tp in self.assignment:
+                self.assignment.remove(tp)
+                revoked.append(tp)
+            else:
+                logger.warn("Could not remove topic partition %s from assignment as it was not previously assigned to %s" % (tp, node.account.hostname))
```

Comment:
do we understand why this situation is happening? Is it related maybe to the mismatch assignment failure we've seen elsewhere in the tests? My point is just to make sure we're not hiding the real failure with this change. I wouldn't expect that the consumer would ever receive a partition to revoke if it was not previously assigned right?

Reply:
You’re right @lianetm, this fix could result in a sweeping the problem under the rug, so to speak. I'll change the logic so that this case still results an error, but with more information so we can debug.

Reply:
@lianetm—I changed the logging to an `assert` that provides useful information for troubleshooting:

```python
tp = _create_partition_from_dict(topic_partition)
assert tp in self.assignment, \
    "Topic partition %s cannot be revoked from %s as it was not previously assigned to that consumer" % \
    (tp, node.account.hostname)
self.assignment.remove(tp)
```

Reply:
thanks! better I believe. Do we have a Jira to investigate the failure leading to this? it's concerning (and even more if the case is that is happening with the new protocol only??)

Reply:
@lianetm—I will file a JIra on this in the next day or two. Thanks!

Reply:
Filed KAFKA-16623, FYI.